### PR TITLE
executor: fix show create table for hash partition

### PR DESCRIPTION
**Pull request by mccxj**
### What problem does this PR solve? 

fixed https://github.com/pingcap/tidb/issues/8415

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8477)

### CODE REVIEW

File: executor/show.go [hunk 1]
```diff
@@ -691,33 +691,38 @@ func (e *ShowExec) fetchShowCreateTable() error {
...
-		if partitionInfo.Columns != nil && partitionInfo.Type == model.PartitionTypeRange {
-			buf.WriteString(fmt.Sprintf("\nPARTITION BY RANGE COLUMNS("))
-			for i, col := range partitionInfo.Columns {
-				buf.WriteString(col.L)
-				if i < len(partitionInfo.Columns)-1 {
-					buf.WriteString(",")
-				}
-			}
-			buf.WriteString(") (\n")
+		if partitionInfo.Type == model.PartitionTypeHash {
```
Comment (winkyao) [inline executor/show.go]: can we use a function to wrap this logic, make code look clear?

**Comment by zz-jason** [comment]
@mccxj Please modify your PR description according to https://help.github.com/articles/closing-issues-using-keywords/, so that after this PR getting merged, the referenced issue can get closed automatically by github.

File: executor/show.go [hunk 2]
```diff
@@ -691,33 +691,38 @@ func (e *ShowExec) fetchShowCreateTable() error {
...
+			buf.WriteString(fmt.Sprintf("\nPARTITION BY HASH( %s )", partitionInfo.Expr))
```
Comment (zz-jason) [inline executor/show.go]: you can use `fmt.Fprintf(buf, ("", xx))`

File: executor/show.go [hunk 3]
```diff
@@ -727,22 +697,56 @@ func (e *ShowExec) fetchShowCreateTable() error {
...
 
 	e.appendRow([]interface{}{tb.Meta().Name.O, buf.String()})
 	return nil
 }
 
+func appendPartitionInfo(partitionInfo *model.PartitionInfo, buf *bytes.Buffer) {
+	if partitionInfo == nil {
+		return
+	}
+	// this if statement takes care of range columns case
```
Comment (ciscoxll) [inline executor/show.go]: Please update the comments.

Comment (mccxj) [reply executor/show.go]: @ciscoxll I have moved comment to the origin position.

File: executor/show.go (hunk 2 above)
Comment (mccxj) [reply executor/show.go]: @zz-jason done.

File: executor/show.go (hunk 1 above)
Comment (mccxj) [reply executor/show.go]: @winkyao done.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @zimulala

**Comment by ciscoxll** [comment]
@zimulala @winkyao PTAL

**Comment by tiancaiamao** [comment]
@crazycs520

**Pull request merged**